### DISCUSSION TITLE
webxr games loading test @ China

### DISCUSSION THREAD

**Discussion by dingsing2021** [hms-ecosystem/OpenXR-SDK#53] [General]
**website：**heyvr.io webxr games
**test net:** china telecom, Home broadband， no VPN to access some famous overseas websites like youtube，google etc.

**here is the test results:**

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/W00386~1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/W00386~1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

  | GAMES/ device & browser | huawei vr glass & wolvic   1.5.2 | Quest3 wolvic 1.5.2 | Oculus browser quest3 | safari@visionOS simulator | Safari-MacBookpro | firefox--MacBookpro
-- | -- | -- | -- | -- | -- | -- | --
1 | Codename: Spectral Dawn     Put your training to good use agent! | <20s | cann't open | cann't open | <5s | <5s | <10s
2 | Frosty     Cut down evil snowmen with your candy cane sword before they melt your mind   … | <5s | cann't open | cann't open | <5s | <5s | <5s
3 | Red and Dead: Zombie   Mayhem     Survive zombie waves in the abandoned Soviet pioneer camp. | <30s | cann't open | cann't open | <5s | <5s | <5s
4 | Archery Evolution: Time   Trials     Unleash precision in a sci-fi space setting. Battle players worldwide,   wield … | <5s | cann't open | cann't open | <5s | <5s | <5s

</body>

</html>

**some key finds**
1, safari is the faster one of loading any webxr games than wolvic ,firefox , and oculus browser .
2, all the webxr games in heyvr.io can be loaded sucess in safari. so i think there is no firewall between;
3, oculus quest 3 + wolvic or oculus browser could not open heyvr.io; -- maybe it is a bug.

**Comment by svillar** [comment]
First of all we need to check whether there is some networking issue. I guess there is none because your other devices show normal load times but let's better discard it. We can get some interesting information from remote debugging. You can check these [instructions](https://github.com/Igalia/wolvic/wiki/Debugging) on how the set it up.

Once you have it working you can select the "network" tab in the inspector window (like in [this one](
![image](https://github.com/open-ecosystem-development/OpenXR-SDK/assets/1826122/603cda5b-9dfa-4919-bdd6-6b5ba72abd15)
) in your PC. That tab records how much time each network request takes. Then go to wolvic in your HVR device and type the url of one of the experiences that take so much time. 

The "network" tab in your desktop Firefox will show the time required for each network requests. If the problem is in the network it'll appear there. It'd be nice if you could then post an screenshot here so we could take a look.

Thanks!

**Reply by dingsing2021** [reply]
thanks I have set up my computer and phone, but this debug tools only support usb connect, but the usb port should be connect to the vr glass. 
so do we have an wifi debug tools?

**Reply by dingsing2021** [reply]
update: the usb connect debug is ok, and I can debug in Firefox pc & Firefox android (for test);

but the wifi debug, the device can be connected ok by wifi:
dingwei@dingdeMacBook-Pro ~ % adb connect 192.168.3.28:6080
already connected to 192.168.3.28:6080
but in the Firefox pc , it shows timeout, 
Connection timed out
“connection  still pending, check for messages on the target browser”
￼--- I have found the solution, and it needs to enable the usb debug.

**Reply by svillar** [reply]
In order to connect via wifi you have to follow these steps
1. Connect device via cable to the PC
2. adb tcpip 5555
3. adb connect IP_OF_THE_DEVICE:5555
4. Unplug the USB cable

The port can be 5555 or any other.

**Reply by dingsing2021** [reply]
thanks , everything is ok.

**Comment by dingsing2021** [comment]
i have try some games(about 20 games) of heyvr.io on the weekend in my home network @ wolvic + Huawei vr glass,  all of them are loading fast(in 10s, some less than 5s), it is interesting, i will do more test. 
all the testing result that i send before all in the office network @ a guest wifi,    our office network provide  a guest wifi and a employee wifi. the guest wifi can access all china websites.
so I Will do more test on different network.
one more thing,  it seems wolvic can cache the webxr content on the disk ,it seems when I open the same webxr content in the second or third times, it loads fast in 5s. please help to confirm it.

**Reply by svillar** [reply]
So perhaps at your offices you have additional firewalls that are making those connections much slower.

With regard to caches, yes you're right, as other browsers we cache the resources on disk to be able to load them faster the next time.

**Reply by dingsing2021** [reply]
yes, I think you are right maybe there are some additional firewalls .

**Comment by JackJohansson** [comment]
@dingsing2021 Hello, developer of heyVR.io here :) 

We have just deployed a CDN system in asia, specifically in China. It should improve the experience of Chinese users a lot, specially after the 2nd phase which is coming soon.

Please let me know if you'd like to test it, have any questions, or would like to investigate matters regarding Chinese visitors. Many thanks!
